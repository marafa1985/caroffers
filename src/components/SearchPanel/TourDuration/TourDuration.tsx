import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import Radio, { RadioProps } from '@material-ui/core/Radio'
import RadioGroup from '@material-ui/core/RadioGroup'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import { Duration } from '../../../model/duration'

interface Props {
    setDuration: (duration: number) => void
    duration: number
}

const GreenRadio = withStyles({
    root: {
        color: '#ff7d19',
        '&$checked': {
            color: '#ff7d19',
        },
    },
    checked: {},
})((props: RadioProps) => <Radio color='default' {...props} />)

const TourDuration: React.FC<Props> = (props) => {
    const { duration, setDuration } = props
    return (
        <RadioGroup style={{ marginLeft: 20 }}
            aria-label='position'
            name='position'
            value={duration}
            onChange={(e) => { setDuration(Number(e.target.value)) }} row>
            {
                Duration.map((durationValue) =>
                    <FormControlLabel key={durationValue.hours}
                        value='top'
                        control={
                            <GreenRadio
                                checked={duration === durationValue.value}
                                onChange={(e) => { setDuration(Number(e.target.value)) }}
                                value={durationValue.value}
                                name='radio-button-demo'
                                inputProps={{ 'aria-label': `${durationValue.value}` }}
                            />
                        }
                        label={`${durationValue.hours} hours`}
                        labelPlacement='end'
                    />
                )
            }
        </RadioGroup>
    )
}

export default TourDuration
