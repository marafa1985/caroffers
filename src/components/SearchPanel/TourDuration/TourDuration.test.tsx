import React from 'react';
import { render } from '@testing-library/react';
import TourDuration from './TourDuration';

test('renders learn react link', () => {
    const component = render(<TourDuration setDuration={() => { }} duration={120} />);
    expect(component).toMatchSnapshot();
});
