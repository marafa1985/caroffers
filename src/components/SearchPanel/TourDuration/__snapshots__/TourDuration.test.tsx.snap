// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="position"
        class="MuiFormGroup-root MuiFormGroup-row"
        role="radiogroup"
        style="margin-left: 20px;"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root Component-root-9 PrivateSwitchBase-checked-17 Mui-checked Component-checked-10"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                aria-label="120"
                checked=""
                class="PrivateSwitchBase-input-19"
                name="radio-button-demo"
                type="radio"
                value="120"
              />
              <div
                class="PrivateRadioButtonIcon-root-32 PrivateRadioButtonIcon-checked-34"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-33"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            2 hours
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root Component-root-9"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                aria-label="180"
                class="PrivateSwitchBase-input-19"
                name="radio-button-demo"
                type="radio"
                value="180"
              />
              <div
                class="PrivateRadioButtonIcon-root-32"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-33"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            3 hours
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root Component-root-9"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                aria-label="240"
                class="PrivateSwitchBase-input-19"
                name="radio-button-demo"
                type="radio"
                value="240"
              />
              <div
                class="PrivateRadioButtonIcon-root-32"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-33"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            4 hours
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="position"
      class="MuiFormGroup-root MuiFormGroup-row"
      role="radiogroup"
      style="margin-left: 20px;"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root Component-root-9 PrivateSwitchBase-checked-17 Mui-checked Component-checked-10"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              aria-label="120"
              checked=""
              class="PrivateSwitchBase-input-19"
              name="radio-button-demo"
              type="radio"
              value="120"
            />
            <div
              class="PrivateRadioButtonIcon-root-32 PrivateRadioButtonIcon-checked-34"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-33"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          2 hours
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root Component-root-9"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              aria-label="180"
              class="PrivateSwitchBase-input-19"
              name="radio-button-demo"
              type="radio"
              value="180"
            />
            <div
              class="PrivateRadioButtonIcon-root-32"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-33"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          3 hours
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root Component-root-9"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              aria-label="240"
              class="PrivateSwitchBase-input-19"
              name="radio-button-demo"
              type="radio"
              value="240"
            />
            <div
              class="PrivateRadioButtonIcon-root-32"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-33"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                />
              </svg>
            </div>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          4 hours
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
